internal/runtime/client/client.go

```diff
@@ -153,6 +157,106 @@ func (c *client) Unregister(extensionConfig *runtimev1.ExtensionConfig) error {
 	return nil
 }
 
+func (c *client) Hook(hook catalog.Hook) HookClient {
+	return &hookClient{
+		client: c,
+		hook:   hook,
+	}
+}
+
+// HookClient is the client used to make calls of the runtime hooks and runtime extension handlers.
+type HookClient interface {
+	// CallAll calls all the extension registered for the hook.
+	CallAll(ctx context.Context, request runtime.Object, response runtimehooksv1.AggregatableResponse) error
+
+	// Call calls only the extension with the given name.
+	Call(ctx context.Context, name string, request, response runtime.Object) error
+}
+
+type hookClient struct {
+	client *client
+	hook   catalog.Hook
+}
+
+func (h *hookClient) CallAll(ctx context.Context, request runtime.Object, response runtimehooksv1.AggregatableResponse) error {
+	gvh, err := h.client.catalog.GroupVersionHook(h.hook)
+	if err != nil {
+		return errors.Wrap(err, "failed to compute GroupVersionHook")
+	}
+	registrations, err := h.client.registry.List(catalog.GroupHook{Group: gvh.Group, Hook: gvh.Hook})
+	if err != nil {
+		return errors.Wrap(err, "failed to retrieve ExtensionHandlers information")
+	}
+	responses := []*runtimehooksv1.ExtensionHandlerResponse{}
+	// Future-work: Call the extension handlers concurrently.
```

Comment:
I would suggest to drop this comment entirely for now. Let's just see if that requirement comes up later. I'm not sure if that optimization is needed.